src/mongo/Config.class.php

```diff
@@ -2026,6 +2064,48 @@ public function getTransactionLogDatabase($readPreference = \MongoClient::RP_PRI
     }
 
     /**
+    * Creates a specification revision subdocument for a composite specification.
+    *
+    * @param \MongoCollection $specification - the composite specification
+    * @return array|null - the subdocument to add to the composite
+    **/
+    public static function getSpecRevisionSubDoc($specification) {
+        if(isset($specification) && isset($specification[_ID_KEY]) && isset($specification[_REVISION])) {
+            return array(
+                _SPEC_KEY => array(
+                    _SPEC_TYPE=> $specification[_ID_KEY],
```

Comment:
It'd probably make more sense if the example I used included the constant:

```php
define('_ID_TYPE','type');
```

that was the inspiration.